libethdrivers/src/plat/zynqmp/io.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#pragma once
+
+#define __arch_getl(addr)         *((volatile uint32_t*)(addr))
+#define __arch_getw(addr)         *((volatile uint16_t*)(addr))
+#define __arch_getb(addr)         *((volatile uint8_t*)(addr))
+
+#define __arch_putl(val, addr)    *((volatile uint32_t*)(addr)) = val
+#define __arch_putw(val, addr)    *((volatile uint16_t*)(addr)) = val
+#define __arch_putb(val, addr)    *((volatile uint8_t*)(addr)) = val
+
+
+//#define __raw_writel(...) writel(__VA_ARGS__)
```

Comment:
consider removing or add a comment why this is commented out.

Reply:
Copied directly from zynq7000 libethdrivers. The only design decisions are in the second commit, where the code was modified to work with the zynqmp.